src/main/java/com/example/shorturl/controller/login/AuthController.java

```diff
@@ -0,0 +1,54 @@
+package com.example.shorturl.controller.login;
+
+import com.example.shorturl.model.user.UserEntity;
+import com.example.shorturl.model.user.UserRequest;
+import com.example.shorturl.model.user.UserResponse;
+import com.example.shorturl.service.user.UserService;
+import com.example.shorturl.utils.exceptions.UserException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import java.util.Objects;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("api/auth")
+public class AuthController {
+
+    private final UserService userService;
+
+    public AuthController(final UserService userService){
+        this.userService = userService;
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<UserResponse> signUp(@Valid @RequestBody UserRequest userRequest) throws UserException {
+
+        if(Objects.isNull(userRequest))
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User request is missing or empty");
+
+        Optional<UserEntity> optionalUser = userService.findUserByUserName(userRequest.getUserName());
```

Comment:
I don't think, this is a good idea to fetch info from the database 2 times.
Let's have some common method with a user request argument, that will check if this is user is presented or not once

Because you are checking by username and then by email. We need to think something else

Reply:
Fixed.